heuristic/repair_operators.py

```diff
@@ -324,228 +218,120 @@ def worst_employee_repair(
         It continues to do so until either the total negative deviation from demand is below a threshold (6)
         or we do not have any employees to assign to this shift as all employees are working this day 
     """
-
-    # print("worst_employee_repair is running")
+    
+    #print("worst_employee_repair is running")
     repair_set = []
-    destroy_set = destroy_set
+    destroy_set = destroy_set.copy()
     employees_changed = employees
-
-    while True:
-        calculate_deviation_from_demand(
-            state, competencies, t_covered_by_shift, employee_with_competencies, demand, destroy_set
-        )
-        shifts = {
-            (t1, v1): -sum(
-                state.soft_vars["deviation_from_ideal_demand"][c, t]
-                for c in competencies
-                for t in t_covered_by_shift[t1, v1]
-            )
-            - v1
-            for t1, v1 in all_shifts
-        }
+    allowed_competency_combinations = {combination for combination in employee_with_competency_combination for item in employee_with_competency_combination[combination] for e in employees_changed if e == item[1]}
```

Comment:
I have never seen this concept before, could you explain it?

Reply:
In respect to code or why I do it this way?